packages/core/src/material/PBRMaterial.ts

```diff
@@ -506,4 +596,29 @@ export class PBRMaterial extends PBRBaseMaterial {
       }
     }
   }
+
+  private _onSpecularColorChanged(): void {
+    const specularColor = this.specularColor;
+    const enableSpecularColor = specularColor.r + specularColor.g + specularColor.b > 0;
+    if (enableSpecularColor !== this._specularColorEnabled) {
+      this._specularColorEnabled = enableSpecularColor;
+      if (enableSpecularColor) {
+        this.shaderData.enableMacro("MATERIAL_ENABLE_SPECULAR_COLOR");
+      } else {
+        this.shaderData.disableMacro("MATERIAL_ENABLE_SPECULAR_COLOR");
+      }
+    }
+  }
+
+  private _updateReflectance(): void {
+    let reflectance: number;
+    if (this.specular) {
```

Comment:
Why the reflectance is this logical? It should be that specularColor and intensity together affect the F0 of the dielectric, and when the default value is 1, F0 should be 0.04